src/pages/WorkspacesList/Rows.tsx

```diff
@@ -0,0 +1,159 @@
+import { formatDate, formatRelativeDate } from '../../services/helpers/date';
+import { buildDetailsPath, toHref, buildIdeLoaderPath } from '../../services/helpers/location';
+
+export interface RowData extends IRow {
+  props: {
+    workspaceId: string;
+  };
+}
+
+export function buildRows(
+  history: History,
+  workspaces: che.Workspace[],
+  deleted: string[],
+  filtered: string[],
+  selected: string[],
+  sortBy: { index: number, direction: SortByDirection }
+): RowData[] {
+  const rows: RowData[] = [];
+  workspaces
+    // skip workspaces that are not match a filter
+    .filter(workspace => filtered.includes(workspace.id))
+    .sort((workspaceA, workspaceB) => {
+      if (sortBy.index === 1) {
+        const nameA = workspaceA.devfile.metadata.name || '';
+        const nameB = workspaceB.devfile.metadata.name || '';
+        return sort(nameA, nameB, sortBy.direction);
+      }
+      if (sortBy.index === 2) {
+        const updatedA = workspaceA.attributes?.updated || workspaceA.attributes?.created || 0;
+        const updatedB = workspaceB.attributes?.updated || workspaceA.attributes?.created || 0;
+        return sort(updatedA, updatedB, sortBy.direction);
+      }
+      return 0;
+    })
+    .forEach(workspace => {
+      const isSelected = selected.includes(workspace.id);
+      const deleting = deleted.includes(workspace.id);
+
+      const pathToDetails = buildDetailsPath(workspace);
+      const linkToDetails = toHref(history, pathToDetails);
+
+      const pathToIde = buildIdeLoaderPath(workspace);
+      const linkToIde = toHref(history, pathToIde);
+
+      try {
+        rows.push(buildRow(workspace, isSelected, deleting, linkToDetails, linkToIde));
+      } catch (e) {
+        console.warn('Skip workspace: ', e);
+      }
+    });
+  return rows;
+}
+
+function sort(a: string | number, b: string | number, direction: SortByDirection): -1 | 0 | 1 {
+  if (a > b) {
+    return direction === SortByDirection.asc ? 1 : -1;
+  } else if (a < b) {
+    return direction === SortByDirection.asc ? -1 : 1;
+  }
+  return 0;
+}
+
+export function buildRow(
+  workspace: che.Workspace,
+  isSelected: boolean,
+  isDeleted: boolean,
+  linkToDetails: string,
+  linkToIde: string
+): RowData {
+  if (!workspace.devfile.metadata.name) {
+    throw new Error('Empty workspace name.');
+  }
+  if (!workspace.attributes) {
+    throw new Error('Empty workspace attributes');
+  }
+  if (!workspace.namespace) {
+    throw new Error('Empty namespace');
+  }
+
+  /* workspace status indicator */
+  const statusIndicator = (<WorkspaceIndicator status={workspace.status} />);
+  /* workspace name */
+  const details = (
+    <span>
+      {statusIndicator}
+      <a href={linkToDetails}>{workspace.devfile.metadata.name}</a>
+    </span>
+  );
+
+  /* last modified time */
+  const { created, updated } = workspace.attributes;
+  const lastModifiedMs = parseInt(updated ? updated : created, 10);
```

Comment:
Something went wrong with my ordering by lastModified:
![Screenshot_20210111_184443](https://user-images.githubusercontent.com/5887312/104211800-31ca1e80-543d-11eb-9ce4-9f4f34d51313.png)

<details>
<summary>The workspaces JSONs I have</summary>

```json
[
  {
    "attributes": {
      "infrastructureNamespace": "sleshche-che",
      "stackName": ".NET Core",
      "created": "1610383137294"
    },
    "namespace": "sleshche",
    "id": "workspace3wkhf9zj6bvd5s02",
    "temporary": false,
    "status": "STOPPED",
    "devfile": {
      "metadata": {
        "name": "dotnet-btuy4"
      }
    }
  },
  {
    "attributes": {
      "stopped": "1610104965579",
      "stoppedAbnormally": "false",
      "stackName": "Apache Camel K",
      "infrastructureNamespace": "sleshche-che",
      "created": "1610103970331",
      "updated": "1610104965535",
      "stopped_by": "activity-checker",
      "org.eclipse.che.runtimes_id": "runtimesyklw5k0dxmem9vna"
    },
    "namespace": "sleshche",
    "id": "workspacedj90f79305js1o3x",
    "temporary": false,
    "status": "STOPPED",
    "devfile": {
      "metadata": {
        "name": "apache-camel-k-utir5"
      }
    }
  },
  {
    "attributes": {
      "stackName": "Apache Camel K",
      "stoppedAbnormally": "false",
      "updated": "1610383123827",
      "infrastructureNamespace": "sleshche-che",
      "org.eclipse.che.runtimes_id": "runtimespe6kykgflhs73siw",
      "created": "1610383123199",
      "stopped": "1610383128570"
    },
    "namespace": "sleshche",
    "id": "workspacestrp5jq22txlzb7r",
    "temporary": false,
    "status": "STOPPING",
    "devfile": {
      "metadata": {
        "name": "apache-camel-k-y7ka6"
      }
    }
  }
]
```

</details>

Reply:
P.S. It can be reproduced with stopping workspace.